src/internal/cnd-handlers.c

```diff
@@ -0,0 +1,52 @@
+#include <rlang.h>
+
+#include "decl/cnd-handlers-decl.h"
+
+r_obj* ffi_try_catch(r_obj* try_catch_args) {
+  r_obj* env = r_node_cadr(try_catch_args);
+
+  r_obj* handlers = KEEP(rlang_env_dots_list(env));
+  r_env_poke(env, rlang_syms.handlers, handlers);
+
+  r_obj* classes = r_names(handlers);
+  if (classes == r_null) {
```

Comment:
This check causes `try_catch(1 + 1)` to error. I think I'd expect it to work.

Reply:
Good catch, this is especially important since we support `!!!` and the handlers list size might vary.